libbpf-rs/src/map.rs

```diff
@@ -59,6 +59,23 @@ impl OpenMap {
     pub fn set_inner_map_fd(&mut self, inner: &Map) {
         unsafe { libbpf_sys::bpf_map__set_inner_map_fd(self.ptr, inner.fd()) };
     }
+
+    /// Reuse a already-pinned map for `self`.
+    pub fn reuse_map<P: AsRef<Path>>(&mut self, path: P) -> Result<()> {
+        let cstring = util::path_to_cstring(path)?;
+
+        let fd = unsafe { libbpf_sys::bpf_obj_get(cstring.as_ptr()) };
+        if fd < 0 {
+            return Err(Error::System(errno::errno()));
+        }
+
+        let ret = unsafe { libbpf_sys::bpf_map__reuse_fd(self.ptr, fd) };
```

Comment:
looking at the code of bpf_map__reuse_fd(), I think you are supposed to close(fd) whether this call succeeded or failed.

Reply:
Funny story: I woke up in the middle of the night and thought "I forgot to close the fd" and when I woke up this morning I forgot until now